src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,73 @@
+package tasks;
+
+import exceptions.DukeException;
+
+
+/** A class that functions as an abstraction of a task. */
+public abstract class Task {
+
+    public static final String UNKNOWN_INPUT_ERROR_STRING = "I don't think I know what this is!";
+    public static final String BAD_DESCRIPTION_ERROR_STRING = "Description cannot be empty!";
+    public static final String INVALID_TASK_NUM_ERROR_STRING = "Not a valid task number!";
+    public static final String TASK_NUM_DOES_NOT_EXIST_ERROR_STRING = "Task %d does not exist!";
+
+    public enum TaskType {
+        TODO,
+        EVENT,
+        DEADLINE
+    }
+    private boolean isDone = false;
+
+    /**
+     * Sets the Task as done or undone.
+     *
+     * @param newDone True if the task is to be set as done, false to be set as undone.
+     */
```

Comment:
I appreciate the extensive use of JavaDoc comments in your code, but I think since this a trivial setter method it does not warrant a JavaDoc comment of its own?